### PR TITLE
Migration plugin: Create/integrate migration progress screen

### LINKED ISSUE #72365: Move terms and conditions above button
See: 
CleanShot 2023-01-20 at 15 26 32@2x

### PR DESCRIPTION
**Pull request by bogiii**
Fixes https://github.com/Automattic/dotcom-forge/issues/1492
Fixes https://github.com/Automattic/wp-calypso/issues/72365
Fixes https://github.com/Automattic/dotcom-forge/issues/1639
Fixes https://github.com/Automattic/dotcom-forge/issues/1641
Fixes https://github.com/Automattic/dotcom-forge/issues/1642
Fixes https://github.com/Automattic/dotcom-forge/issues/1643

## Proposed changes:

* Installed `@automattic/jetpack-api` dependency
* Added fetchMigrationStatus API method
* Added ToS link above the Get started button
* Created Migration Progress component
* Integrated Migration Progress into the Migration plugin
* Introduced MigrationError component and handled case when import is in progress

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Launch a [JN site with the migration plugin](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-migration=origin/update/migration-plugin-2&wp-debug-log).
* Check if there is `ToS` link above the `Get Started` button
* Click on the `Get Started` button
* Start import everything
* Check if the migration plugin change it's screen to the Migration Progress (see screenshot below)
* Check if the `Check your migration progress` button redirects to the proper `migrationHandler` route

## Screenshots
Screenshot 2023-02-07 at 10 39 11

error-component

Screenshot 2023-02-09 at 15 54 33

Screenshot 2023-02-10 at 23 46 54

### CODE REVIEW

**Comment by ouikhuan** [comment]
It seems to only fetch the migration status once when the page is loaded. Migration might be done with the screen opened, but if we only fetch the status on page load, the screen won't be updated once it's done. 

Should we consider calling the migration status every few seconds like what we do on WordPress.com now? So if the migration is done the screen will update accordingly?

**Review by ouikhuan — DISMISSED** [review]
LGTM!

**Comment by bogiii** [comment]
Hey @ouikhuan,

Could you please review it once again? In the meantime, I added:
- hook for fetch status in intervals of 3 seconds
- term of service above the "Get started" button

The PR description is updated as well. Remember that I branched out from the https://github.com/Automattic/jetpack/pull/28770 PR since this change depends on that part. I'll fix conflicts once we merge the first PR.

Thanks!

File: projects/plugins/migration/src/js/components/hooks/use-migration-status.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useCallback } from 'react';
+import { MigrationStatus } from '../migration/types';
+
+/**
+ * Get migration status
+ *
+ * @param {object} restApi - Configured restApi
+ * @returns {MigrationStatus} - MigrationStatus object
+ */
+export function useMigrationstatus( restApi ): MigrationStatus {
```
Comment (ouikhuan) [inline projects/plugins/migration/src/js/components/hooks/use-migration-status.ts]: Could we also check the users' connection first and decide to fetch it or not? If user hasn't connect to WordPress.com site yet, it's probably not efficiency to have it check every three second. What do you think?

Comment (bogiii) [reply projects/plugins/migration/src/js/components/hooks/use-migration-status.ts]: For sure, that’s a good point. I’ll add that part. Thanks!

**Review by ouikhuan — APPROVED** [review]
Looks great to me, thanks for the great work @bogiii !

**Review by jeherve — APPROVED** [review]

**Pull request merged**